main.tf

```diff
@@ -186,6 +186,23 @@ resource "aws_s3_bucket" "default" {
             replica_kms_key_id = try(rules.value.destination.replica_kms_key_id, null)
             account_id         = try(rules.value.destination.account_id, null)
 
+            # https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-walkthrough-5.html
+            dynamic "metrics" {
+              for_each = try(rules.value.destination.metrics.status, "") == "Enabled" ? [1] : []
+
+              content {
+                status = "Enabled"
+              }
+            }
+
+            dynamic "replication_time" {
+              for_each = try(rules.value.destination.metrics.status, "") == "Enabled" ? [1] : []
+
+              content {
+                minutes = 15
+              }
```

Comment:
This looks as if we are hardcoding it. We should add an inline comment clarifying that it is not. Also, we need to set the `enabled` attribute here.

```suggestion
              content {
                status = "Enabled"
                # we are not hardcoding this value — rather, this is the only value accepted by the provider, either implicitly or explicitly.
                minutes = 15
              }
```

Reply:
minutes are ignored here

correct syntax is 

```
metrics {
  event_threshold {
    minutes = 15
  }
  status = "Enabled"
}
```
but it's not supported by the latest aws provider, I don't know why

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_replication_configuration

Reply:
the link I posted has this
https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-walkthrough-5.html

![image](https://user-images.githubusercontent.com/5246026/151792814-3aa681d7-5a80-4dfe-94c0-b11ec0ec01eb.png)

I don't want to replicate their limitations in code

Honestly i'm waiting until they re-release this whole api as it's a total nonsense

Reply:
It does not make sense to build any further around this until they redo the api